src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioInfoBarService.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [Export(typeof(IInfoBarService))]
+    internal class VisualStudioInfoBarService : IInfoBarService
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly IForegroundNotificationService _foregroundNotificationService;
+        private readonly IAsynchronousOperationListener _listener;
+
+        [ImportingConstructor]
+        public VisualStudioInfoBarService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            IForegroundNotificationService foregroundNotificationService,
+            [ImportMany] IEnumerable<Lazy<IAsynchronousOperationListener, FeatureMetadata>> asyncListeners)
+        {
+            _serviceProvider = serviceProvider;
+            _foregroundNotificationService = foregroundNotificationService;
+            _listener = new AggregateAsynchronousOperationListener(asyncListeners, FeatureAttribute.InfoBar);
+        }
+
+        public void ShowInfoBarInActiveView(string message, params InfoBarUI[] items)
+        {
+            ShowInfoBar(activeView: true, message: message, items: items);
+        }
+
+        public void ShowInfoBarInGlobalView(string message, params InfoBarUI[] items)
+        {
```

Comment:
add assert that this can be called from any thread (available on ForegroundThreadAffinitizedObject).